corehq/apps/export/export.py

```diff
@@ -458,6 +459,8 @@ def should_rebuild_export(export, last_access_cutoff):
     :return: True if export should be rebuilt
     """
     # Don't rebuild exports that haven't been accessed since last_access_cutoff or aren't enabled
+    if isinstance(export, six.string_types):
+        export = get_properly_wrapped_export_instance(export)
```

Comment:
This concerns me somewhat from a performance standpoint - previously it fetched all saved export ids and spun each up into a new celery task.  Now it fetches all export ids and then immediately does individual gets for each export document.

The simplest fix for this would probably be to use `iter_docs` to get the export documents in chunks.

An ideal solution would be to include `last_accessed` in `export_instances_by_is_daily_saved/view` so we can avoid the lookup entirely.  That couch view is only used to support the `get_all_daily_saved_export_instance_ids` dbaccessor, so it shouldn't be hard to change.   Then `last_access_cutoff` could be a parameter to the dbaccessor, which could filter exports at the database level.

Reply:
There aren't many documents but can be definitely improved.
Thank you.

Reply:
added the bool and timestamp in the couch view and filtered on that.
one thing i am not sure about is that if this would also include the exports which does not have timestamp set. Please help me confirm if the view is correctly added.

Reply:
As-is, I believe the view will emit exports which don't have a timestamp set, but the timestamp field will be `null` or `undefined`.  Since this is the only usage of that view, you can really rewrite it as much as you like to fit this use-case.  I'll echo Simon's comments regarding the view too, that is, you can wrap the `emit` statement in an `if` block to only return daily saved exports potentially eligible for automatic rebuilding.

If a field isn't going to be used as a parameter in a view call, then it doesn't need to be part of the emit statement.  For instance, we're currently storing an index for this view for all export documents, then filtering out ones which aren't daily saved exports at query time.  If we instead make that a check in the mapping (`if doc.is_daily_saved_export {emit...}`), then we can make the index smaller by only storing daily saved exports.  You're of course correct that at this scale there's probably not much of a concern for performance there, though.

One final note - happily, [this view lives in the `meta` db](https://github.com/dimagi/commcare-hq/blob/b715a0566ed72b49ee52c8f88fe760101e05e690/corehq/couchapps/__init__.py#L10-L10), meaning that the reindex should be very fast.

Reply:
thank you.

I tested this locally that the docs with empty value for last_accessed were not fetched using `startkey` but could iterate over the values though would be great to do this via search itself if there is a way.

Updated view.

Reply:
Is your goal to always include docs with an empty value for `last_accessed`?

The easiest way to do that would probably be to make two separate queries - one to retrieve docs where `last_accessed > whatever`, and a second to retrieve those where `last_accessed === None`.  To do that, you can pass in `key=[None]` rather than `startkey` and `endkey`.

Couch may have some provision to combine both of those queries into one request, though I'm not aware of a way to do that, and two requests seems fine.

Reply:
fair enough. Made the change.